src/components/updateInput.jsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from "react";
+import { account, databases } from "../appwrite/config"; 
+import { Modal, Box, TextField, Button } from "@mui/material";
+
+const UpdateInput = ({ type, open, handleClose, userData }) => {
+  const [inputValue, setInputValue] = useState("");
+  const [password, setPassword] = useState("");
+  const [oldPassword, setOldPassword] = useState("");
+
+  const handleUpdate = async () => {
+    try {
+      switch (type) {
+        case "email":
+          if (!password) {
+            alert("Please enter your password to update your email.");
+            return;
+          }
+          await account.updateEmail(inputValue, password);
+          await databases.updateDocument(
+            import.meta.env.VITE_APPWRITE_DB_ID,
+            import.meta.env.VITE_APPWRITE_ORG_COLLECTION_ID, 
+            userData.$id, 
+            { email: inputValue }
+          );
+          alert("Email updated successfully!");
+          location.reload();
```

Comment:
Page ko reload kar dena optimised hai to nahi, for now rehne dete hai, baad me students ke liye bhi yahi method hi use hoga, to isko modify karna hi padega tab isko ContextAPI me dal denge, fir wahi se sab ho jaega